jsonurl.d.ts

```diff
@@ -0,0 +1,64 @@
+interface JsonURLOptions {
+    // These are defined as protected in the comments
+    // setOrDefault(src, key, defValue);
+    // setOverride(src, key);
+    // setOrDefaultInt(src, key, defValue);
+}
+
+export interface JsonURLParseOptions extends JsonURLOptions {
+    //See https://github.com/jsonurl/jsonurl-js/blob/master/src/JsonURLParseOptions.js
+
+    //public properties
+    maxParseDepth?: number; //default 1 << 5 === 32
+    maxParseValues?: number; //default 1 << 12 === 4096
+    maxParseChars?: number; //default 1 << 15 === 32K
+    emptyValue?: any; //default is {}
+    nullValue?: any; //default is null
+
+    //These feel like they are intended to be protected
+    // impliedArray?: any[];
+    // impliedObject?: Record<string, any>;
+    // wwwFormUrlEncoded?: boolean;
+    // impliedStringLiterals?: boolean;
+    // allowEmptyUnquotedValues?: boolean;
+    // allowEmptyUnquotedKeys?: boolean;
+    // coerceNullToEmptyString?: boolean;
+    // noEmptyComposite?: boolean;
+    // AQF?: boolean;
+    // getMissingValue?: (key, pos) => any; //default throws an error
+}
+
+declare class JsonURL {
+    //Source comments suggest this is deprecated:
```

Comment:
It was deprecated very recently. I left tests in place because the next release will be a semver minor increment. I'll likely remove it for 2.0